src/attestation.rs

```diff
@@ -592,24 +592,31 @@ pub(crate) fn verify_apple_anonymous_attestation(
     let x5c_array_ref =
         cbor_try_array!(x5c_value).map_err(|_| WebauthnError::AttestationStatementX5CInvalid)?;
 
+    let credential_public_key = crypto::COSEKey::try_from(&acd.credential_pk)?;
+    let alg = credential_public_key.type_;
+
     let arr_x509: Result<Vec<_>, _> = x5c_array_ref
         .iter()
-        .map(|values| {
+        .enumerate()
+        .map(|(i, values)| {
+            let alg = if i == 0 {
```

Comment:
I was hoping for something better like this, so thanks for the suggestion. Your interpretation of the logic is indeed correct.

Part of the reason I didn't spend too much time optimizing this is that I honestly have no idea what COSE content type is at play with the remaining certs in the chain. The spec doesn't give any guidance and clearly it's not necessarily the same as the attestation cert.

The one clue I have is that the Apple webauthn root certificate uses ECDSA_384, which I ascertained by tossing it into a python repl and peering in. This leads me to wonder:

1. Is there some convention that would suggest that certs in a trust path are all encoded in the same way?
2. Is there a particular reason we must specify the COSE content type ahead of time rather than decoding and inferring? This probably reflects a lack of understanding on my part but it would seem to me that we can parse an X509 object and ascertain the encoding from the result, as I did in python with the Apple webauthn root.

Reply:
Only the credentialPublicKey is COSE, the values in x5c should all be base64 encoded x.509 certificates.

Reply:
Got it—I'm then surprised then that this code works at all, especially given the comment above the `try_from` implementation
> // Must be DER bytes. If you have PEM, base64decode first!

As a follow-on, should I take this to mean that the code suggestion above and the code used to decode the `x5c` in the tpm attestation format should not be decoding under any `COSEContentType`.

Reply:
I think they are all DER, not PEM. If it wasn't this would be failing and your tests would be failing.

Reply:
This makes sense to me, but leaves me confused about the correct choice of `COSEContentType` when decoding the cert chain beyond the attestation cert.

If these needs to be addressed, then my understanding is that it probably does too in the TPM attestation format, and thus another PR is needed regardless and so I don't have an issue merging this for now. However, I do want to understand what's going on here a bit better.

Reply:
I think the current approach is fine, because we know that the other certs in the chain are this type. If it breaks then we'll have to look at a different approach. It can sometimes be better to be "static/fragile" like this because if anything does change we'll "fail noisy" and we can then work out what is going on. If we made this generic over all possible types we may actually miss some other important change in the process. That's why I'm happy for this to remain "static" wrt to the cosecontent type here.

Reply:
That sounds good to me. Should I delete the FIXME comment here, just the FIXME bit, or leave the comment as is?

Reply:
I'd leave it as is :)